js/MarkupSubmissionsBatchConversion.js

```diff
@@ -0,0 +1,177 @@
+			$.pkp.plugins.markup.js.MarkupSubmissionsBatchConversion.prototype.processFile();
+		});
+	};
+
+	/**
+	 * A Number, representing the ID value of the timer that is set.
+	 * @private
+	 * @type {int}
+	 */
+	$.pkp.plugins.markup.js.MarkupSubmissionsBatchConversion.timer = null;
+
+	/**
+	 * A jQuery element.
+	 * @private
+	 * @type {jQueryObject}
+	 */
+	$.pkp.plugins.markup.js.MarkupSubmissionsBatchConversion.$listElement = null;
+
+	/**
+	 * Determine which file to convert
+	 */
+	$.pkp.plugins.markup.js.MarkupSubmissionsBatchConversion.prototype.processFile = 
+		function() {
+			var $next = $('input[class=submission-file]:not(".processed"):checked').first();
+			if ($next.length) {
+				this.convert($next);
+			}
+	}
+
+	/**
+	 * Trigger file conversion
+	 * 
+	 * @param {jQueryObject} $element input checkbox representing submission file
+	 */
+	$.pkp.plugins.markup.js.MarkupSubmissionsBatchConversion.prototype.convert = 
+		function($element) {
+			this.$listElement = $element.closest('li');
+			this.$listElement.addClass('batch-processing');
+			var params = {
+				'submissionId': $element.data('submission-id'),
+				'stage': $element.data('stage'),
+				'fileId': $element.data('file-id'),
+			};
+			var postUrl = $('input#conversionTriggerUrl').val() +
+							'?submissionId='+$element.data('submission-id')+
+							'&stage='+$element.data('stage')+
+							'&fileId='+$element.data('file-id')+
+							'&target=galley-generate';
+
+			var that = this;
+			$.post(postUrl, {}, function(data) {
+				that.$listElement.append(data.content);
+				$('.pkp_spinner', that.$listElement).addClass('is_visible');
+				var jobId = that.$listElement.find('span#conversionJobId').text();
+				if ((jobId != "") && (that.timer == null) ) {
+					that.timer = setInterval(function() { that.fetchJobStatus.apply(that); }, 5000);
+				}
+			}, 'json');
+	}
+
+	/**
+	 * Callback to fetch xml job status
+	 */
+	$.pkp.plugins.markup.js.MarkupSubmissionsBatchConversion.prototype.fetchJobStatus = 
+		function() {
+			var self = this;
+			$.ajax({
+				url: $('span#conversionJobStatus', this.$listElement).data('url'),
+				type: 'POST',
+				dataType: 'json',
+				success: function(data) {
+					if (data['content'].hasOwnProperty('status') && data['content'].hasOwnProperty('isCompleted')) {
+						if (data['content']['isCompleted']) {
+							self.stopAndClear();
+							$('span#conversionJobStatus').text(data['content']['status']);
+							self.$listElement.removeClass('batch-processing').addClass('batch-success');
+							$('input[class=submission-file]', self.$listElement).addClass('processed');
+							self.processFile();
+						}
+						else {
+							$('span#conversionJobStatus', self.$listElement).text(data['content']['status']);
+						}
+					}
+				},
+				error: function() {
+					self.$listElement.removeClass('batch-processing').addClass('batch-failure');
+					$('span#conversionJobStatus', self.$listElement).text('An unexpected error occured.');

```

Comment:
Is some more specific message available? This will be hard to debug.

Reply:
this ajax request could fail for multiple reasons. 

Would  the message below be more appropriate? 
`"An unexpected error occured while fetching OTS job status during batch conversion."`

Reply:
Will this be e.g. a failure in the conversion of the document (e.g. a file format problem or something user-caused), or a failure in the service (e.g. downtime)?

Reply:
this is a simple ajax post request to fetch some json response from a URL. So it can range from server not being reachable to lack of internet on the user side.

Reply:
OK, gotcha. This is OK as it is, I think.